test/dynamo/test_aot_autograd.py

```diff
@@ -352,8 +354,11 @@ def guard_fail_fn(failure):
             failure_reason = failure[0]
 
         fxy = torch._dynamo.optimize(cc, guard_fail_fn=guard_fail_fn)(F())
-        fxy(x, y)
-        fxy(x, y)
+        # Note: to prevent a recompilation between the two calls,
+        # we need to clone x and y on each use.
+        # fxy mutates the input's metadata, so otherwise dynamo will end up recompiling.
+        fxy(x1, y1)
+        fxy(x2, y2)
```

Comment:
@ezyang I think this PR actually fixed a bug that I didn't notice. In this test, we were originally asserting that if we called fxy(x, y) twice with the same inputs, we wouldn't recompile. But actually, we **should** recompile. `fxy()` mutates the strides on its input, and so the next time we call `fxy()`, the tensor metadata has changed - since we're running with static shapes, we'd expect to recompile.

I didn't carefully diagnose why my PR fixed this case, but it was caught in this test. I updated it to clone inputs so that that we properly don't recompile on the first invocation (and added a separate test below)